File: app/src/main/java/com/blues/module_gank/old/model_ios/view/IOSImagesAdapter.java
package com.blues.module_gank.old.model_ios.view;

import static com.blues.constant.ConstantKt.IMAGE_URL;

import android.app.Activity;
import android.content.Intent;
import android.text.TextUtils;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.blues.R;
import com.blues.framework.utils.ScreenUtil;
import com.blues.framework.utils.SizeUtil;
import com.blues.module_gank.old.ShowImageActivity;
import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

import androidx.annotation.Nullable;
import androidx.core.app.ActivityOptionsCompat;

public class IOSImagesAdapter extends BaseQuickAdapter<String, BaseViewHolder> {


    public IOSImagesAdapter(@Nullable List<String> data) {
        super(R.layout.item_item_image, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, String url) {

        //设置每张图片的长度
        int width = ScreenUtil.getScreenWidth();
        ViewGroup.MarginLayoutParams params = new ViewGroup.MarginLayoutParams(width / 4, SizeUtil.dp2px(150));
        params.setMargins(SizeUtil.dp2px(5), SizeUtil.dp2px(5), SizeUtil.dp2px(5), 0);
        helper.getView(R.id.iv_item).setLayoutParams(params);

        if (url != null && !TextUtils.isEmpty(url)) {
            Glide.with(helper.getView(R.id.iv_item).getContext()).load(url).placeholder(R.drawable.shape_place_holder).into((ImageView) helper.getView(R.id.iv_item));

            helper.getView(R.id.iv_item).setOnClickListener(v -> {
                Intent intent = new Intent(mContext, ShowImageActivity.class);
                intent.putExtra(IMAGE_URL, url);
                ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation((Activity) mContext, helper.getView(R.id.iv_item), "transitionImage");
                mContext.startActivity(intent, options.toBundle());
            });
        }
    }
}

File: app/src/main/java/com/blues/gankio/v1/view/IOSImagesAdapter.kt
package com.blues.gankio.v1.view

import com.blues.framework.utils.ScreenUtil.screenWidth
import com.chad.library.adapter.base.BaseQuickAdapter
import com.blues.R
import android.view.ViewGroup.MarginLayoutParams
import android.text.TextUtils
import android.content.Intent
import androidx.core.app.ActivityOptionsCompat
import android.app.Activity
import android.view.View
import android.widget.ImageView
import coil.load
import com.blues.constant.IMAGE_URL
import com.blues.framework.utils.dp
import com.chad.library.adapter.base.BaseViewHolder

class IOSImagesAdapter(data: List<String?>?) :
    BaseQuickAdapter<String?, BaseViewHolder>(R.layout.item_item_image, data) {

    override fun convert(helper: BaseViewHolder, url: String?) {

        //设置每张图片的长度
        val width = screenWidth
        val params = MarginLayoutParams(width / 4, 150f.dp.toInt()).apply {
            setMargins(5f.dp.toInt(), 5f.dp.toInt(), 5f.dp.toInt(), 0)
        }
        helper.getView<View>(R.id.iv_item).layoutParams = params

        url?.let {
            if (!TextUtils.isEmpty(url)) {

                (helper.getView<View>(R.id.iv_item) as ImageView).load(url){
                    placeholder(R.drawable.shape_place_holder)
                    error(R.drawable.shape_place_holder)
                }

                //Glide.with(helper.getView<View>(R.id.iv_item).context).load(url)
                //    .placeholder(R.drawable.shape_place_holder)
                //    .into((helper.getView<View>(R.id.iv_item) as ImageView))

                helper.getView<View>(R.id.iv_item).setOnClickListener {
                    val intent = Intent(mContext, ShowImageActivity::class.java).putExtra(IMAGE_URL, url)
                    val options = ActivityOptionsCompat.makeSceneTransitionAnimation((mContext as Activity), helper.getView(R.id.iv_item), "transitionImage")
                    mContext.startActivity(intent, options.toBundle())
                }
            }
        }
    }
}